src/pinocchio/setup.rs

```diff
@@ -95,17 +162,24 @@ pub fn setup(qap: Qap, toxic_waste: ToxicWaste) -> EvaluationKey {
     }
 
     EvaluationKey {
-        gv_ks,
-        gw_ks,
-        gy_ks,
-        gv_alphaks,
-        gw_alphaks,
-        gy_alphaks,
+        gv_ks: gv_ks_mid,
+        gw_ks: gw_ks_mid,
+        gy_ks: gy_ks_mid,
+        gv_alphaks: gv_alphaks_mid,
+        gw_alphaks: gw_alphaks_mid,
+        gy_alphaks: gy_alphaks_mid,
         g_s_i,
-        g_beta,
+        g_beta: g_beta_mid,
     }
 }
 
+pub fn setup(qap: Qap, toxic_waste: ToxicWaste) -> (EvaluationKey, VerifyingKey) {
+    (
+        generate_evaluation_key(&qap, toxic_waste),
+        generate_verifying_key(&qap, toxic_waste),
```

Comment:
This two functions call `let g = GroupType::generator();` inside them and compute their hidings in terms of the result of that. Right now that's fine because the `generator` method is deterministic and returns the same in both cases. But if we ever decide to return a random generator, things are gonna brake. I would rather add `g` as a third parameter to the `generate_xxx_key` functions and define `g` in `setup`.

Reply:
I've changed both functions to take the generator as a parameter